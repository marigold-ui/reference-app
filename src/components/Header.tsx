import { SiteLogo } from './SiteLogo';

function Header() {
  return (
    <header className="sticky top-0 z-10 flex h-[--page-header-height] w-full items-center gap-2 border-0 border-b border-solid border-gray-300 bg-white px-[--page-padding] md:px-[--page-padding-md] xl:px-[--page-padding-xl]">
      <div className="hidden gap-6 md:flex md:flex-1">
        <SiteLogo />
      </div>
    </header>
  );
}

export default Header;
