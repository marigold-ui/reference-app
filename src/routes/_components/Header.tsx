import { Link, useRouter } from '@tanstack/react-router';
import { Logo } from '../../components/Logo';

const unkebabify = (val: string) =>
  val
    .split('-')
    .map(word => word.charAt(0).toUpperCase() + word.slice(1))
    .join(' ');

const Navigation = () => {
  const router = useRouter();

  // Get all first level routes
  const routes = Object.keys(router.routesByPath).filter(route =>
    /**
     * ^\/: Asserts that the string starts with a forward slash.
     * [^\/]+$: Ensures that the rest of the string contains one or more characters
     *          that are not forward slashes, and this continues until the end of the string.
     */
    /^\/[^/]+$/.test(route)
  );

  return (
    <nav className="flex items-center gap-4">
      {routes.map(route => (
        <Link
          key={route}
          to={route}
          className="text-sm font-medium text-secondary-600 hover:text-secondary-900 lg:px-1"
        >
          {unkebabify(route.replaceAll('/', ''))}
        </Link>
      ))}
    </nav>
  );
};

export const Header = () => (
  <header className="sticky top-0 z-10 flex h-[--page-header-height] w-full items-center justify-between gap-2 border-0 border-b border-solid border-secondary-200 bg-bg-body px-[--page-padding] md:justify-normal md:px-[--page-padding-md] xl:px-[--page-padding-xl]">
    <Link className="md:pr-8" to="/">
      <Logo className="size-6" />
    </Link>
    <Navigation />
  </header>
);
