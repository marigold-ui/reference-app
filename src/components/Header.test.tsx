import { expect } from 'vitest';
import Header from './Header';
import {
  createRootRoute,
  createRouter,
  RouterProvider,
} from '@tanstack/react-router';
import { act, render } from '@testing-library/react';

test('renders <Header /> correctly', async () => {
  const rootRoute = createRootRoute({
    component: () => <Header />,
  });

  const router = createRouter({
    routeTree: rootRoute,
  });

  const { container } = render(<RouterProvider router={router} />);

  await act(async () => {
    await new Promise(resolve => setTimeout(resolve, 1000));
  });

  expect(container).toMatchSnapshot();
});
