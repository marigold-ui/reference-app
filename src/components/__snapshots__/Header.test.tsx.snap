// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders <Header /> correctly 1`] = `
<div>
  <header
    class="sticky top-0 z-10 flex h-[--page-header-height] w-full items-center gap-2 border-0 border-b border-solid border-gray-300 bg-white px-[--page-padding] md:px-[--page-padding-md] xl:px-[--page-padding-xl]"
  >
    <div
      class="hidden gap-6 md:flex md:flex-1"
    >
      <a
        aria-current="page"
        class="flex items-center gap-1.5 xl:w-[--page-sub-nav-absolute-width] active"
        data-status="active"
        href="/"
      >
        <svg
          class="size-6"
          viewBox="0 0 160 160"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M137 1c-4 0-9 2-13 6-14 14-14 43-14 43s29 0 43-14c4-4 6-8 7-12a77 77 0 00-23-23z"
            fill="#46505a"
          />
          <path
            d="M63 4c-3 3-4 8-4 14 0 20 20 40 20 40s20-20 20-40c0-6-1-11-4-14a80 80 0 00-32 0zm32 154c3-4 4-8 4-14 0-20-20-40-20-40s-21 20-21 40c0 6 2 10 5 14a80 80 0 0032 0zm61-93c-4-3-8-5-14-5-20 0-40 21-40 21s20 20 40 20c6 0 10-1 14-4a80 80 0 000-32zM2 97c4 3 8 4 14 4 20 0 40-20 40-20S36 60 16 60c-6 0-10 2-14 4a80 80 0 000 33zm143 27c-1-4-3-9-7-13-14-14-43-14-43-14s0 29 14 43c4 4 9 6 13 7a80 80 0 0023-23zM13 38c1 4 3 8 7 13 14 14 43 14 43 14s0-29-14-43c-4-4-9-6-13-7a77 77 0 00-23 23zm23 109c4-1 9-3 13-7 14-14 14-43 14-43s-29 0-43 14c-4 4-6 9-7 12a80 80 0 0023 24z"
            fill="#f98422"
          />
        </svg>
        <div
          class="hidden pl-2 text-lg font-bold uppercase text-[#46505a] lg:block"
        >
          Reference App
        </div>
      </a>
    </div>
  </header>
</div>
`;
